matching_brackets.exs

```diff
@@ -0,0 +1,49 @@
+defmodule BracketsBalance do
+  @brackets %{
+    "{" => "}",
+    "(" => ")",
+    "[" => "]"
+  }
+
+  def add_to_stack(stack, counter, check_string) do
```

Comment:
In general, you should try to avoid using indexes in elixir whenever possible - most of operations can be expressed on lists without indexes.
Here, you could remove counter, have a single argument check_string that can be splitted into `[current_symbol | rest_string] = check_string`.
Then you use current_symbol wherever you were doing `Enum.at(check_string, counter)` and then pass rest_string into recursion, stopping it if there is no elements left.